emoji/src/main/java/com/vanniktech/emoji/Utils.java

```diff
@@ -50,19 +45,73 @@
   }
 
   static int dpToPx(@NonNull final Context context, final float dp) {
-    return (int) (dp * context.getResources().getDisplayMetrics().density);
+    return Math.round(TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, dp,
+        context.getResources().getDisplayMetrics()) + 0.5f);
+  }
+
+  static int getOrientation(final Context context) {
+    return context.getResources().getConfiguration().orientation;
   }
 
-  static boolean shouldOverrideRegularCondition(@NonNull final Activity context, final EditText editText) {
+  static boolean shouldOverrideRegularCondition(@NonNull final Context context, final EditText editText) {
     if ((editText.getImeOptions() & EditorInfo.IME_FLAG_NO_EXTRACT_UI) == 0) {
-      return context.getResources().getConfiguration().orientation == Configuration.ORIENTATION_LANDSCAPE;
+      return getOrientation(context) == Configuration.ORIENTATION_LANDSCAPE;
     }
 
     return false;
   }
 
-  static int getScreenHeight(@NonNull final Activity context) {
-    return dpToPx(context, context.getResources().getConfiguration().screenHeightDp);
+  @SuppressWarnings({"unchecked", "JavaReflectionMemberAccess"}) static int getInputMethodHeight(final Context context, final View rootView) {
+    try {
+      final InputMethodManager imm = (InputMethodManager) context.getApplicationContext().getSystemService(Context.INPUT_METHOD_SERVICE);
+      final Class inputMethodManagerClass = imm.getClass();
+      final Method visibleHeightMethod = inputMethodManagerClass.getDeclaredMethod("getInputMethodWindowVisibleHeight");
+      visibleHeightMethod.setAccessible(true);
+      return (int) visibleHeightMethod.invoke(imm);
+    } catch (NoSuchMethodException exception) {
```

Comment:
nit: can be grouped together

Reply:
No they can't, tried that and the IDE complained, because this gets compiled to catch an exception only available on Java 8 if I recall correctly. Catching generic `Exception` should also not be done since it hides `RuntimeException`s on which we want to crash.